### PR TITLE
Update module error message and 2 suffix allowance

### LINKED ISSUE #235: Update Module Validity and error message
* Error Message for valid modules (accept capital letters)
* Modules found to be have 2 suffix: `ASP1201CH`

##### _Based on this, should be allowed to fix in 1.4_
image

### PR DESCRIPTION
**Pull request by whitesnowx**
Plausible module: `ASP1201CH`
--> Increase suffix allowed to increase scope for valid inputs

Closes #235

### CODE REVIEW

File: src/main/java/staffconnect/model/person/Module.java [hunk 1]
```diff
@@ -11,8 +11,8 @@ public class Module {
 
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Module code should contain 2-4 capital letters followed by 4 digits long and at most 1 capitalised suffix";
-    public static final String VALIDATION_REGEX = "[a-zA-Z]{2,4}\\d{4}[a-zA-Z]{0,1}";
+            "Module code should contain 2-4 letters followed by 4 digits long and at most 2 capitalised suffix";
```
Comment (iynixil) [inline src/main/java/staffconnect/model/person/Module.java]: Should it be `and suffix of at most 2 characters`?

File: src/test/java/staffconnect/model/person/ModuleTest.java [hunk 2]
```diff
@@ -31,14 +31,15 @@ public void isValidModule() {
...
         assertFalse(Module.isValidModule("CS210310101010T")); // too many numbers
         assertFalse(Module.isValidModule("CS21T")); // too little numbers
 
         // valid module code
         assertTrue(Module.isValidModule("CS2103")); // 2 prefix, 4 letters without 1 optional suffix
         assertTrue(Module.isValidModule("CS2103T")); // 2 prefix, 4 letters with 1 optional suffix
+        assertTrue(Module.isValidModule("CS2103TT")); // 2 prefix, 4 letters with 2 optional suffix
         assertTrue(Module.isValidModule("GEN2050")); // 3 prefix, 4 letters without  1 optional suffix
         assertTrue(Module.isValidModule("GEN2050Y")); // 3 prefix, 4 letters with  1 optional suffix
         assertTrue(Module.isValidModule("GESS1035")); // 4 prefix, 4 letters without 1 optional suffix
```
Comment (iynixil) [inline src/test/java/staffconnect/model/person/ModuleTest.java]: nit: Misleading comment `4 letters` -> `4 digits`?

**Review by iynixil — COMMENTED** [review]
lgtm! minor nits

File: src/test/java/staffconnect/model/person/ModuleTest.java (hunk 2 above)
Comment (whitesnowx) [reply src/test/java/staffconnect/model/person/ModuleTest.java]: amended 🔧

File: src/main/java/staffconnect/model/person/Module.java (hunk 1 above)
Comment (whitesnowx) [reply src/main/java/staffconnect/model/person/Module.java]: amended 🔧

**Review by iynixil — APPROVED** [review]

**Pull request merged**